### DISCUSSION TITLE
Text, and color detection, + detection counter

### DISCUSSION THREAD

**Discussion by LarchyEverything** [Nain57/Smart-AutoClicker#341] [General Feature Requests]
Text and color detection would be useful, could cut away from having to zoom really far in to get a good image or color. A detection counter would be nice so you can count how many times a condition has activated, so you could disable or enable conditions based off this. Maybe a macro recorder that gets broken down into conditions? But I also would really like a way to close apps because on some android devices, can't use the smart autoclicker to do that. I'm thinking a makeshift way is an app killer app, that you can open, then use the SAC (smart autoclicker) to then terminate X apps and then could be used to also restart them. This is assuming when you start the app with SAC, it automatically takes you to it.

**Comment by LarchyEverything** [comment]
Also two more ideas, a timed condition, how long something needs to be on screen before it activates. So a quick blip wouldn't activate it, but it'd need to be on screen for like 5 seconds. Another quality of life feature would be folders. You can put conditions into folders, and maybe be able to disable an entire folder. Cloning folders would be nice too. This would also be needed for a macro recorder. Since it'd convert all movements and stuff into several conditions so being able to just have that all put into a folder would be a lot more convenient.

**Reply by Cyber5imon** [reply]
Most of this functionality now exists in the new [Beta 3.00](https://github.com/Nain57/Smart-AutoClicker/discussions/421) version. Download [here](https://github.com/Nain57/Smart-AutoClicker/releases/tag/3.0.0-beta04).
